images/router/haproxy/reload-haproxy

```diff
@@ -15,7 +16,40 @@ if [ -f $pid_file ]; then
 fi
 
 if [ -n "$old_pid" ]; then
+  if $(set | grep DROP_SYN_DURING_RESTART= > /dev/null) && [ -n "$DROP_SYN_DURING_RESTART" ]; then
+    # We install the syn eater so that connections that come in during the restart don't
+    # go onto the wrong socket, which is then closed.  The sleep is needed to let the socket
+    # drain before the new haproxy starts and binds to the same port.
+    ports=$(grep -E '^\s*bind\s+:[[:digit:]]+\w' "$config_file" | cut -f2 -d: | paste -d, -s)
+    if [ -n "$ports" ]; then
+      # If this doesn't insert, we don't care, we still want to reload
+      /usr/sbin/iptables -I INPUT -p tcp -m multiport --dports $ports --syn -j DROP \
+			 -m comment --comment "Eat SYNs while reloading haproxy" || :
+      installed_iptables=1
+      sleep 0.1
```

Comment:
Is there another way to determine when it's safe to restart haproxy?  Seemingly arbitrary timeouts always make me worry about the possibility of race conditions.